[CI] ShardBulkInferenceActionFilterIT testBulkOperations failing

You probably just need to apply https://github.com/elastic/elasticsearch/pull/108372 to `internalClusterTest` as well. Although, then it might be easier to just apply Mark's suggestion here: https://github.com/elastic/elasticsearch/pull/108954#discussion_r1611997163.

**Build scan:**
https://gradle-enterprise.elastic.co/s/iyfhixzgrnmg6/tests/:x-pack:plugin:inference:internalClusterTest/org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT/testBulkOperations


**Reproduction line:**
```
./gradlew ':x-pack:plugin:inference:internalClusterTest' --tests "org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT.testBulkOperations" -Dtests.seed=3030EE4E2FF5B0A9 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=da -Dtests.timezone=Asia/Brunei -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT#testBulkOperations`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testBulkOperations'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: No handler for type [semantic_text] declared on field [dense_field]

  at __randomizedtesting.SeedInfo.seed([3030EE4E2FF5B0A9:291CBA0780DDF0BE]:0)
  at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:580)
  at org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:541)
  at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:428)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:1342)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$2(MetadataCreateIndexService.java:473)
  at org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:719)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:471)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:594)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:421)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:300)
  at org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:551)
  at org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1050)
  at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1015)
  at org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:233)
  at org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1656)
  at org.elasticsearch.action.ActionListener.run(ActionListener.java:433)
  at org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1653)
  at org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1248)
  at org.elasticsearch.action.ActionListener.run(ActionListener.java:433)
  at org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1227)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
  at java.lang.Thread.run(Thread.java:1570)

  Caused by: org.elasticsearch.index.mapper.MapperParsingException: No handler for type [semantic_text] declared on field [dense_field]

    at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:334)
    at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:265)
    at org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:452)
    at org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:122)
    at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:578)
    at org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:541)
    at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:428)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:1342)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$2(MetadataCreateIndexService.java:473)
    at org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:719)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:471)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:594)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:421)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:300)
    at org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:551)
    at org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1050)
    at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1015)
    at org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:233)
    at org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1656)
    at org.elasticsearch.action.ActionListener.run(ActionListener.java:433)
    at org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1653)
    at org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1248)
    at org.elasticsearch.action.ActionListener.run(ActionListener.java:433)
    at org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1227)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.lang.Thread.run(Thread.java:1570)

```

--- a/x-pack/plugin/inference/build.gradle
+++ b/x-pack/plugin/inference/build.gradle
@@ -41,7 +41,7 @@ dependencies {
 }
 
 if (BuildParams.isSnapshotBuild() == false) {
-  tasks.named("test").configure {
+  tasks.withType(Test).configureEach {
     systemProperty 'es.semantic_text_feature_flag_enabled', 'true'
   }
 }

--- a/x-pack/plugin/inference/src/internalClusterTest/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilterIT.java
+++ b/x-pack/plugin/inference/src/internalClusterTest/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilterIT.java
@@ -54,7 +54,6 @@ public class ShardBulkInferenceActionFilterIT extends ESIntegTestCase {
         return Arrays.asList(Utils.TestInferencePlugin.class);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/109005")
     public void testBulkOperations() throws Exception {
         Map<String, Integer> shardsSettings = Collections.singletonMap(IndexMetadata.SETTING_NUMBER_OF_SHARDS, randomIntBetween(1, 10));
         indicesAdmin().prepareCreate(INDEX_NAME)

